Chore: add GH action to run tests automatically

mashazyu opened the pull request:
## Description

- Current GH action should run when PR is open/updated.
- Readme updated to keep in mind to update GH action, when updating Dart/Flutter for this project.

## Testing

I tested this action on my repository, so I hope it works for the current repo as well.
https://github.com/mashazyu/learningdart/actions/runs/13993890097/job/39184058000

## Other considerations
- Automatic build on Android and iOS. I first thought about adding this aciton as we, but then I realized that builds actually take a long time and actually maybe not relevant for PRs. Maybe it would be useful to automatically build app and push builds to TestFlight, for example, when branch is merged into main. But maybe this is something for later use/for a set of testers of the app.
- Do you think it would be appropriate to add quick templates for a new PR/Issue along with this change?

felixerdy reviewed (changes requested):
- Based on the action error it seems like there is an issue because of the missing `.env` file. Please try to create a simple `.env` file in the action script
- Is it neccessary to use a `macOS` runner, or can we also use an `ubuntu` runner and run the tests on an Android Emulator? Just asking since the `macOS` runner usually is more expensive.

felixerdy commented:
> * Automatic build on Android and iOS. I first thought about adding this aciton as we, but then I realized that builds actually take a long time and actually maybe not relevant for PRs. Maybe it would be useful to automatically build app and push builds to TestFlight, for example, when branch is merged into main. But maybe this is something for later use/for a set of testers of the app.

Would be cool to have it but not worth the effort right now. Would see it as future work

> * Do you think it would be appropriate to add quick templates for a new PR/Issue along with this change?

That'd be cool to have 👍

mashazyu commented:
> * Based on the action error it seems like there is an issue because of the missing `.env` file. Please try to create a simple `.env` file in the action script

Opps, I missed it. Thank you for bringing this up! I added .env file mock directly in the action code, didn't use gh secrets at the moment, since to run the job no real data is needed.

> * Is it neccessary to use a `macOS` runner, or can we also use an `ubuntu` runner and run the tests on an Android Emulator? Just asking since the `macOS` runner usually is more expensive.

I tried to switch to ubuntu + [Android simulator ](https://github.com/marketplace/actions/android-emulator-runner). It looks like isar mocking should be done differently (if possible) for this configuration, so I switched back to macOs + iOS. I tried some isar mocking solutions that didn't require actual isar service refactoring, but they didn't work for me. So I think the next step would be to try to refactor the service, which I feel is outside of the scope of the current task (please let me know, if you think differently). And I documented possible mocking solutions in the test, which was failing on Android :)

felixerdy reviewed (approved):
Great work, thank you @mashazyu 🚀
